pkg/virt-handler/vm.go

```diff
@@ -550,12 +598,23 @@ func (d *VirtualMachineController) processVmShutdown(vmi *v1.VirtualMachineInsta
 			return err
 		}
 	}
-	d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.Deleted.String(), "VirtualMachineInstance stopping")
 
 	return d.processVmCleanup(vmi)
```

Comment:
We should remove this call to processVmCleanup() now as part of the restructuring. 

I know this logic was here before your patch. Because the cleanup phase removes the watchdog file, I think we should only perform cleanup once the domain has actually been observed as deleted. It was probably a mistake that I tried to couple destroy/delete with cleanup previously. This could also be a reason for why we see stale domains.

Reply:
makes sense